packages/app/api-common/src/paginationUtils.ts

```diff
@@ -201,3 +201,22 @@ export async function getPaginatedEntriesByHasMore<T extends Record<string, unkn
 
   return resultArray
 }
+
+export async function traversePaginatedSourceByHasMore<
```

Comment:
🟢 Could we add JSDoc as in the other methods? just to explain clearly how the method works and how it is intended to be used